### PR TITLE
Replace Travis with GitHub actions

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR replaces Travis CI with GitHub actions, with the `codespell` spellcheck added (produces warnings only).

The following secrets need to be set up by an admin of this repo:
- `AWS_REGION` (same region as the `transitive.js` bucket defined in `package.json`)
- `AWS_ACCESS_KEY_ID`,
- `AWS_SECRET_ACCESS_KEY`,
- ~`GITHUB_TOKEN`~ `GH_TOKEN`,
- `NPM_TOKEN`

To see the GitHub actions execution results for the course of this PR, please refer to checks ran on my fork at https://github.com/binh-dam-ibigroup/transitive.js/pull/1.

### CODE REVIEW

File: .github/workflows/node-ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      # (AWS CLI v2 does not come with a non-sudo installer as of writing).
+      - name: Set up Python (latest 3.x)
+        if: ${{ env.IS_PUSH_TO_MASTER }}
+        uses: actions/setup-python@v2
+      - name: Install pip
+        if: ${{ env.IS_PUSH_TO_MASTER }}
+        run: pip install -U pip
+      - name: Install AWS CLI v1
+        if: ${{ env.IS_PUSH_TO_MASTER }}
+        run: pip install awscli
```
Comment (landonreed) [inline .github/workflows/node-ci.yml:58]: Looks like the AWS CLI is already installed on Ubuntu: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#cli-tools

So, we shouldn't need this step or the ones prior (python install, pip install).

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     "prepublish": "mastarm prepublish lib:build",
     "cover": "npm test -- --coverage",
     "lint": "mastarm lint lib",
-    "semantic-release": "semantic-release pre && npm publish && semantic-release post",
+    "semantic-release": "semantic-release",
```
Comment (landonreed) [inline package.json:33]: It looks like this package script wasn't being used in the Travis CI run. Is that your understanding? I don't fully know what these removed steps used to do, but perhaps @evansiroky can weigh in on whether they're important.

**Review by landonreed — CHANGES_REQUESTED** [review]
Generally looks good, but I do have questions about the change in semantic-release script and whether the AWS CLI v2 can be used.

File: package.json (hunk 2 above)
Comment (binh-dam-ibigroup) [reply package.json:33]: Correct, Travis was invoking `semantic-release` directly, which fails with GH actions (unless I missed something).

File: .github/workflows/node-ci.yml (hunk 1 above)
Comment (binh-dam-ibigroup) [reply .github/workflows/node-ci.yml]: Good find, and it looks like AWS CLI v2 is preinstalled on there, so it saves on execution time (see https://github.com/conveyal/transitive.js/pull/63/commits/2a826d6ead65a21adf2ebb6990814ee2b8bdb44b)

**Review by landonreed — APPROVED** [review]

**Review by evansiroky — APPROVED** [review]

**Comment by evansiroky** [comment]
@binh-dam-ibigroup, I went ahead and merged this, but the deployment of storybook didn't get triggered (see https://github.com/conveyal/transitive.js/runs/2172632643?check_suite_focus=true). Can you try to figure out why?

**Comment by binh-dam-ibigroup** [comment]
> @binh-dam-ibigroup, I went ahead and merged this, but the deployment of storybook didn't get triggered (see https://github.com/conveyal/transitive.js/runs/2172632643?check_suite_focus=true). Can you try to figure out why?

It appears the example code from GitHub I was originally using was incorrect :(. I fixed it in https://github.com/conveyal/transitive.js/pull/65.

**Pull request merged**